.github/workflows/ruff.yml

```diff
@@ -12,5 +12,5 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-      - run: pip install --user ruff
-      - run: ruff check --output-format=github .
+      - uses: astral-sh/setup-uv@v4
+      - run: uvx ruff check --output-format=github .
```

Comment:
This is unrelated to this PR but maybe we should combine the "ruff.yml" workflow into a more general "lint.yml" where there are other linters as well in addition to Ruff and they can utilize the "lint" dependency group.

Reply:
So...  This is a good question.  Ruff is run both here and in pre-commit.  Here it is run with `--output-format=github` to give contributors clear GH annotations and in `pre-commit` without '--output-format'.  Is there a way to run it just once on each PR but give GitHub Annotations on PRs but not on localhost?

Reply:
Hmm, that's a good point. Does pre-commit support the GitHub annotation output format? If it does, we could just utilize that otherwise I'm not sure what to do. You might be interested in the new `astral-sh/ruff-action@v2` which is a complete rewrite.